content/tutorials/react-native-presence-publish-subscribe.textile

```diff
@@ -0,0 +1,539 @@
+
+In the @submitName@ method, you used @AsyncStorage@ to set the value of @user@ which you are checking for in the @componentDidMount@ method to the value of the current @txt@ state.
+
+Just after the @submitName@ method, you defined the @handler@ method assigned in the constructor. In this method, you set the state of @user@ to null. Why have I done this?  Is it needed? You will find out in the next method called @getRender@
+
+In the @getRender@ function, you first check if the @user@ state is not null, then return the @Chat@ component which is still empty for now. If the @user@ state is null, it means the user is yet to select a user name. If the @user@ state is null, it presents them with a form to enter their user name.
+
+Take a look at the form been returned when there is no user name, you will notice:
+
+# The presence of a @TextInput@ whose value is set to the @txt@ state and the @onChangeText@ event alters the @txt@ state
+# The presence of a @Button@ whose @onPress@ event fires the @submitName@ event you had defined earlier.
+
+If you run the app, you will be presented with a screen that shows a text input and button prompting you to set a user name.  
+
+<a href="/images/tutorials/react-native/enter-username-react-native-presence-publish-subscribe.png" target="_blank">
+  <img src="/images/tutorials/react-native/enter-username-react-native-presence-publish-subscribe.png" style="width: 100%" alt="enter username screenshot">
+</a>
+
+Alternatively, on iOS, this is what it looks like:
+
+<a href="/images/tutorials/react-native/enter-username-react-native-presence-publish-subscribe-ios.png" target="_blank">
+  <img src="/images/tutorials/react-native/enter-username-react-native-presence-publish-subscribe-ios.png" style="width: 100%" alt="enter username screenshot">
+</a>
+
+However, if you set a user name now, you will be presented with an empty page, as you have not created the chat component yet. Open @chat.js@ and replace the content with:
+
+```[javascript]
+
+/**
+ * Sample React Native App
+ * https://github.com/facebook/react-native
+ * @flow
+ */
+import React, { Component } from 'react';
+import {
+  StyleSheet,
+  Text,
+  View,
+  TextInput,
+  Button,
+  ScrollView,
+  AsyncStorage
+} from 'react-native';
+
+
+var realtime = require('ably').Realtime;
+var ably, channel;
+
+export default class Chat extends Component<{}> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: '',
+      msg: [],
+      txt: ''
+    } 
+  }
+
+  componentDidMount = () => {
+    AsyncStorage.getItem("user").then((value) => {
+            this.setState({"user": value});
+            this.subscribe();
+        }).done();
+  }
+
+  subscribe = () =>{
+    ably = new realtime({ key:'XXX_APP_ID', clientId: this.state.user });
+    channel = ably.channels.get('cog-tie');
+    channel.presence.subscribe('enter', (member)=> {
```

Comment:
It does not appear you are receiving `present` members i.e. all members who are already present at the time you enter this channel.  You are only catching new people who enter or leave and thus not syncing all members.

Reply:
Please read the docs at https://www.ably.io/documentation/realtime/presence for more info on syncing the members on a channel

Reply:
Here, i have added an update that uses the other memebers data, showing how many members are currently in the chat room